app/assets/javascripts/annotations.js

```diff
@@ -471,7 +529,75 @@ function newAnnotationFormCode() {
     }
 
 
-    submitNewAnnotation(comment, shared_comment, score, problem_id, line, $(this));
+    submitNewAnnotation(comment, shared_comment, false, score, problem_id, line, $(this));
+  });
+
+  return box;
+}
+
+// Create form code to create / update a global annotation
+// config takes the following keys: problem, score (for edit), comment (for edit)
+function globalAnnotationFormCode(newAnnotation, config) {
+  const box = $(".base-global-annotation-form").clone();
+  box.removeClass("base-global-annotation-form");
+
+  if (!newAnnotation) {
+    box.find(".comment").val(config.comment);
+    box.find(".score").val(config.score);
+    box.find('input[type=submit]').val("Update annotation");
+  }
+
+  var problemNameToIdMap = {};
+  _.each(problems, function (problem) {
+    problemNameToIdMap[problem.name] = problem.id;
+  })
+
+  box.find('.annotation-form').show();
+  box.find('.annotation-cancel-button').click(function (e) {
+    e.preventDefault();
+    $(this).parents(".annotation-form").parent().remove();
+  })
+
+  box.find('#comment-textarea').autocomplete({
+    appendTo: box.find('#comment-textarea').parent(),
+    minLength: 0,
+    delay: 0,
+    source: localCache["shared_comments"]
+  }).focus(function () {
+    $(this).autocomplete('search', $(this).val())
+  });
+
+  box.tooltip();
+
+  box.find('.annotation-form').submit(function (e) {
+    e.preventDefault();
+    var comment = $(this).find(".comment").val();
+    var shared_comment = $(this).find("#shared-comment").is(":checked");
+    var score = $(this).find(".score").val();
+    var problem_id = problemNameToIdMap[config.problem];
+
+    if (comment === undefined || comment === "") {
+      box.find('.error').text("Annotation comment can not be blank!").show();
+      return;
+    }
+
+    if (score === undefined || score === "") {
+      box.find('.error').text("Annotation score can not be blank!").show();
+      return;
+    }
+
+    if (newAnnotation) {
+      submitNewAnnotation(comment, shared_comment, true, score, problem_id, 0, $(this));
+    } else {
+      const annotationObject = getAnnotationObject(config.annotationId);
```

Comment:
nit: same thing as prev nit about newAnnotation object

Reply:
Updated to use `Object.assign` instead